lib/dashboard/lessons/lesson_editor.ex

```diff
@@ -176,6 +176,12 @@ defmodule UneebeeWeb.Live.Dashboard.LessonEditor do
     end
   end
 
+  @impl Phoenix.LiveView
+  def handle_info({LessonEdit, :lesson_edit, updated_socket}, socket) do
+    %{lesson: lesson} = updated_socket.assigns
+    {:noreply, assign(socket, lesson: lesson)}
+  end
```

Comment:
Even though this approach works, it adds some extra code that might be unnecessary because [we're navigating back to the page using push_patch](https://github.com/zoonk/uneebee/blob/d0ec9c696f06e509f59ffa71571b7d207a38ef42/lib/dashboard/lessons/components/lesson_edit.ex#L60). This means the [handle_params called is going to be called](https://github.com/zoonk/uneebee/blob/d0ec9c696f06e509f59ffa71571b7d207a38ef42/lib/dashboard/lessons/lesson_editor.ex#L30) and we already have the updated lesson data there because [we call Content.list_lessons/1](https://github.com/zoonk/uneebee/blob/d0ec9c696f06e509f59ffa71571b7d207a38ef42/lib/dashboard/lessons/lesson_editor.ex#L34).

Therefore, I suggest that we use this `lessons` list to find the updated lesson and assign it back to our socket:

```elixir
updated_lesson = Enum.find(lessons, fn l -> l.id == lesson.id end)
```

Then, add this to the socket assignments, you can just add a new line with the updated lesson

```elixir
assign(:lesson, updated_lesson)
```

What do you think? Does that make sense?

Reply:
@wceolin This is an amazing review I have to go back and retrace my step, I agree with you using `push_patch` to navigate after saving the changes will push the updated state to the client without reloading the whole page and this makes the updated lesson object available on the client side. Thank you for the suggestions